apollo-portal/src/main/java/com/ctrip/framework/apollo/openapi/v1/controller/BranchController.java

```diff
@@ -0,0 +1,139 @@
+    @RequestMapping(value = "/apps/{appId}/clusters/{clusterName}/namespaces/{namespaceName}/branches/{branchName}/operator/{operator}/merge", method = RequestMethod.POST)
+    public ReleaseDTO merge(@PathVariable String appId, @PathVariable String env,
+                            @PathVariable String clusterName, @PathVariable String namespaceName,
+                            @PathVariable String branchName, @PathVariable String operator, @RequestParam(value = "deleteBranch", defaultValue = "true") boolean deleteBranch,
+                            @RequestBody NamespaceReleaseModel model, HttpServletRequest request) {
+
+        if (model.isEmergencyPublish() && !portalConfig.isEmergencyPublishAllowed(Env.fromString(env.toUpperCase()))) {
+            throw new BadRequestException(String.format("Env: %s is not supported emergency publish now", env));
+        }
+
+        ReleaseDTO createdRelease = namespaceBranchService.merge(appId, Env.valueOf(env.toUpperCase()), clusterName, namespaceName, branchName,
+                model.getReleaseTitle(), model.getReleaseComment(),
+                model.isEmergencyPublish(), deleteBranch, operator);
+
+        ConfigPublishEvent event = ConfigPublishEvent.instance();
+        event.withAppId(appId)
+                .withCluster(clusterName)
+                .withNamespace(namespaceName)
+                .withReleaseId(createdRelease.getId())
+                .setMergeEvent(true)
+                .setEnv(Env.valueOf(env.toUpperCase()));
+
+        publisher.publishEvent(event);
+
+        return createdRelease;
+    }
+
+
+    @RequestMapping(value = "/apps/{appId}/clusters/{clusterName}/namespaces/{namespaceName}/branches/{branchName}/rules", method = RequestMethod.GET)
+    public GrayReleaseRuleDTO getBranchGrayRules(@PathVariable String appId, @PathVariable String env,
+                                                 @PathVariable String clusterName,
+                                                 @PathVariable String namespaceName,
+                                                 @PathVariable String branchName) {
+
+        return namespaceBranchService.findBranchGrayRules(appId, Env.valueOf(env.toUpperCase()), clusterName, namespaceName, branchName);
+    }
+
+
+    @PreAuthorize(value = "@consumerPermissionValidator.hasModifyNamespacePermission(#request, #appId, #namespaceName)")
+    @RequestMapping(value = "/apps/{appId}/clusters/{clusterName}/namespaces/{namespaceName}/branches/{branchName}/operator/{operator}/rules", method = RequestMethod.PUT)
+    public void updateBranchRules(@PathVariable String appId, @PathVariable String env,
+                                  @PathVariable String clusterName, @PathVariable String namespaceName,
+                                  @PathVariable String branchName, @PathVariable String operator, @RequestBody GrayReleaseRuleDTO rules,
```

Comment:
normally, operator should be a RequestParam(query parameter) instead of a PathVariable